scadalts-ui/.gitignore

```diff
@@ -29,5 +29,7 @@ test-results.xml
 *.sw?
 
 node_modules/*
-!@min-gb/
-
+!@min-gb/dist
+!@min-gb/public
+!@min-gb/src
+!@min-gb/*.*
```

Comment:
We don't want it in ignore, this library is used in our code, and since it is no longer available in the remote repository we have moved it to sources, what lies ahead is to integrate this library in our sources:

Reply:
I'm aware of that, thats why i put the !, to include it in the repo. I changed it to that to ignore the node_modules folder inside @min-gb

Reply:
Hi @YuaFox, 'to include it in the repo'? 
I mean what did you do because I don't understand...

Reply:
Okay, I understand, you made it so that it won't be ignored? Great, we need to audit this library, what is it, etc.

Reply:
From what I can see, this configuration does not block the visibility of changes in gb-min, so there is no need to change it here, unless I am missing something, I just did a test and changed the file: \node_modules\@min-gb\vuejs-components\babel. config.js and git sees this change.

Reply:
However, this is interesting because in the old gitignore file we do not ignore node_modules from scadalts-ui and yet the content of node_modules is ignored... if so, there is no point in changing it, although it is worth considering what causes this, it is probably some pattern from gitignore file...

Reply:
@YuaFox  " I changed it to that to ignore the node_modules folder inside @min-gb" What is the justification for this? Because I don't remember it causing any problems...

Reply:
Show me on screenshot..

Reply:
We get errors like this, but it looks like the new front is working...
[node_logs.txt](https://github.com/user-attachments/files/16797512/node_logs.txt)

Theoretically, entries regarding not supported for a given library are disturbing. Especially regarding the Node engine.
One might wonder whether this somehow affects the final compilation of the new fronted.

From what I see, the change to ignore is not necessary unless you show me something else.

Reply:
![image](https://github.com/user-attachments/assets/23d74e14-af5a-46f5-9957-6126ad51a721)

Reply:
With your gitignore corrections:
![image](https://github.com/user-attachments/assets/31ba24e1-66bb-43ff-8ddb-f6aa9c9d5ea6)

Reply:
A lot of libraries Scada-LTS is using reached EOL including Vue 2.7. Upgrading to Vue 3 and removing deprecated libraries for new ones removes all warnings.

I can do it, but it will take time.

Reply:
Vue 3 is compatible with vue 2 components, but there are some things that change. For example, this.$router no longer works and you have to do const router = useRouter(). I'll open an issue later about this topic and check if its a big change. There's a migration tool too.

All unit tests are working (Mocha and Java JUnit) but cypress is not, but i think those test are outdated right? I'm getting 404 errors to urls like http://localhost:8080/Scada-LTS/modern_watch_list.shtm but this test fails on the current release.

Reply:
http://localhost:8080/Scada-LTS/modern_watch_list.shtm 
You can delete this case, we no longer have this page.

Reply:
Some are failing because things like this
`CypressError: `cy.server()` was removed in Cypress version 12.0.0. Please update to use `cy.intercept()` instead.`

But thats not a problem, its just a function name change.

The rest are failing for things that no longer exist ej. Create Datasource fails because "Data source has been saved" is no longer displayed. (I'm testing this on a scada-lts 2.7.8 release too). I dont think that cypress tests are reliable in its current state.

Reply:
@YuaFox,
1. You check release/2.8.0, which is the version we are currently working on.
2. 'But thats not a problem, its just a function name change.' What will you do about it?
3. Create Datasource fails because "Data source has been saved" is no longer displayed? Where? 
If it was displayed, it should still be like that, something must have gone broken...

Reply:
Hello @Limraj 
2. This commit fixes the deprecated function https://github.com/SCADA-LTS/Scada-LTS/pull/2990/commits/b87af6884c6cf04e72fca3f3d08ccd8b5a0f8b7a
3. I attach the cypress tests summary (I ran scadaLTS 2.7.8 to compare it with this pr)
[cypress_testing.txt](https://github.com/user-attachments/files/16815434/cypress_testing.txt) 
Yep, some things have gone broken, or changed but not reflected on tests.

I can fix the tests or fix features that are expected to work but I'll need some help :(

Reply:
@YuaFox,
You need to check whether sometimes these tests did not break down on the old node. As for the operation of functions in the application, I think it is enough that the frontend compilation process runs successfully, then Node has nothing to do with the operation of the application itself.

Reply:
Hello @Limraj 
Yes you're correct, Node has nothing to do with the operation of the application, scada-lts is only using it to compile the frontend, but i think cypress tests should be fixed and enforced to run correctly before merging. Maybe we should fix this in another task?

Reply:
@YuaFox You just need to make sure whether there were problems with them on node 14 as well. If so, they can be corrected in a separate task.

Reply:
Hi @YuaFox,
Can you confirm that the tests worked similarly on node 14? 
If so, I'll take a look at the code and we'll merge.

Regards,
Kamil Jarmusik

Reply:
Hi @Limraj ,

If you have these files changed feel free to to commit them here.

The unit tests work as expected, the cypress tests there are some that worked on node 14 but not on node 22. I haven't had time to take a closer look and see if it's just a change in the dom or something similar or if they are real bugs.

Reply:
Hi @YuaFox,
usage command:
`npx cypress run`
![image](https://github.com/user-attachments/assets/9437b679-ecf9-4b7f-ae6e-76429fbee41d)
then:
![image](https://github.com/user-attachments/assets/d1273c9f-188e-407d-b8d0-f22c49928a50)